<launch>
  <arg name="mav_name"                default="euroc3"/>
  <arg name="namespace"               default="$(arg mav_name)" />
  <arg name="usb_port"                default="/dev/ttyUSB1"/>
  <arg name="dxl_baud_rate"           default="1000000"/>
  <arg name="manipulator_name"        default="wp_manipulator_3r"/>

  <arg name="use_moveit"              default="true"/>
  <arg name="use_joint_state"         default="true"/>
  <arg name="use_cmd_vel"             default="false"/>

  <group ns="$(arg namespace)" >
    <node name="mav_interface" pkg="ethzasl_mav_interface" type="mav_interface_node"
      output="screen" respawn="true" clear_params="true">
      <param name="frame_id" value="fcu" />
      <param name="publish_nav1_imu" value="true" />
      <rosparam file="$(find euroc_launch)/parameters/$(arg mav_name)/mav_interface.yaml"/>
      <param name="capability_group" value="Core" />
    </node>

    <node name="mav_linear_mpc" pkg="mav_linear_mpc" type="mav_linear_mpc_node"  respawn="true" clear_params="true" output="screen">
      <remap from="odometry" to="msf_core/odometry" />
      <param name="control_interface/verbose" value="true"/>
      <param name="K_x" value="0.0" />
      <param name="Kf_x" value="0.9" />
      <param name="g_x" value="9.81" />
      <param name="K_y" value="0.0" />
      <param name="Kf_y" value="0.9" />
      <param name="g_y" value="9.81" />
      <rosparam file="$(find euroc_launch)/parameters/common/linear_mpc_neo.yaml"/>
      <param name="capability_group" value="Core" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="world_tf_broadcaster" args="0 0 0 0 0 0 1 world odom 100">
      <param name="capability_group" value="Core" />
    </node>

    <!-- Vicon -->
    <node name="pose_sensor_vicon" pkg="msf_updates" type="pose_sensor" respawn="true" clear_params="true" output="screen">
      <remap from="msf_updates/transform_input" to="vrpn_client/estimated_transform" />
      <remap from="msf_core/imu_state_input" to="imu" />
      <rosparam file="$(find euroc_launch)/parameters/common/msf_parameters_vicon.yaml"/>
      <param name="capability_group" value="Vicon" />
    </node>

    <node name="vrpn_client" type="ros_vrpn_client" pkg="ros_vrpn_client" output="screen">
      <param name="vrpn_server_ip" value="optitrack" />
      <param name="vrpn_coordinate_system" value="optitrack" />
      <param name="object_name" value="$(arg mav_name)" />
      <param name="vicon_estimator/dt" value="0.01" />
      <param name="translational_estimator/kp" value="1.0" />
      <param name="translational_estimator/kv" value="10.0" />
      <param name="rotational_estimator/orientation_estimate_initial_covariance" value="1" />
      <param name="rotational_estimator/rate_estimate_initial_covariance" value="1" />
      <param name="rotational_estimator/orientation_process_covariance" value="0.01" />
      <param name="rotational_estimator/rate_process_covariance" value="1" />
      <param name="rotational_estimator/orientation_measurementCovariance" value="0.0005" />
      <param name="capability_group" value="Vicon" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="vicon_tf_broadcaster" args="0 0 0 0 0 0 1 world vicon 100">
      <param name="capability_group" value="Vicon" />
    </node>

    <node pkg="rosserial_python" name="serial_node" type="serial_node.py" args="/dev/ttyUSB1"/>
    
  </group>
</launch>